Doc/library/venv.rst

```diff
@@ -172,6 +172,10 @@ creation according to their needs, the :class:`EnvBuilder` class.
 
     * ``upgrade_deps`` -- Update the base venv modules to the latest on PyPI
 
+    * ``gitignore`` -- a Boolean value which, if true, will create a
```

Comment:
Late to the party, but as maintainer of the virtualenv project, wouldn't a better API be to make this an 
`scm_ignore: set[str] | None = None` that defaults for now to `{"git"}`. That way we allow people to extend EnvBuilder to support their own SCM... and we're not locking it down to gitignore only.

Reply:
Let's pretend that it's 10 years from now and svn becomes popular again. What would that look like? I assume we would add an `svnignore` keyword argument like now. At some point we may change the defaults, but even with the `scm_ignore_files` that would occur. I think the question is are we okay with potentially adding a new keyword roughly every 8 years when Python changes what version control system it supports (it's 32 years old publicly now and has gone through CVS, svn, hg, and now git)?

The other question I have is whether your concern is only for the programmatic API, or does this extend to the CLI as well? Do you care about `--without-gitignore`, or did you expect to introduce an `--without-scm-ignore-file`?

Reply:
> The other question I have is whether your concern is only for the programmatic API, or does this extend to the CLI as well? Do you care about `--without-gitignore`, or did you expect to introduce an `--without-scm-ignore-file`?

The later. I'd imagine one could do `--without-scm-ignore-file git hg`.

> think the question is are we okay with potentially adding a new keyword roughly every 8 years when Python changes what version control system it supports (it's 32 years old publicly now and has gone through CVS, svn, hg, and now git)?

My worry is a different here. I want the API to be open for extension. Sure hg might not be popular in general; however, it might be the go to thing at Facebook (or insert other random company). So I'd prefer to offer the option for that company to add a plugin/extension where they can add support for their favorite SCM flavour even if upstream is not interested in supporting at the moment. My biggest stake in getting an API that's extensible is that whatever we add here virtualenv would have to mirror it. And while adding plugins to the stdlib is difficult, so allowing extensability might be low on priority, it's much more realistic for virtualenv. Then as a bonus we'd not have to add a new flag every time we add a new SCM system at some point down the line, and we'd also not be a breaking change when we decide to drop things. Overall I don't see many downsides of going down this path.

Reply:
Thanks for the feedback! It seems compelling, but I don't want to switch to it without thinking about it a little bit.

Reply:
- [ ] You're welcome, do let me know what you decide so I can follow it up on virtualenv side. Also would be good to have a programmatic way of finding out if this is available on the venv side or not (could be a CPython minor.patch). We should likely also ping PYPY for awareness then.

Reply:
Programmatically you could either do it based on the Python version -- magic of being a part of the stdlib 😁 -- or by using `inspect` to check for the parameter. But since this is a feature it would be a check against the major.minor version which will be 3.13 (as I assume _something_ will get merged and I don't think this will take until May 😉).